File: test-suite-java/src/test/java/io/micronaut/el/test/TextFunctions.java
package io.micronaut.el.test;

import io.micronaut.el.annotation.ELFunction;

public final class TextFunctions {

    private TextFunctions() {
    }

    public static int length(String value) {
        return value == null ? 0 : value.length();
    }

    @ELFunction("upper")
    public static String toUpperCase(String value) {
        return value == null ? null : value.toUpperCase();
    }
}

File: doc-examples/example-kotlin/src/test/kotlin/example/TextFunctions.kt
package example

import io.micronaut.el.annotation.ELFunction

object TextFunctions {

    @JvmStatic
    @ELFunction // <1>
    fun shout(text: String): String = text.uppercase() + "!"

    @JvmStatic
    @ELFunction("initials") // <2>
    fun initialsOf(text: String): String = text.split(" ").joinToString("") { it.substring(0, 1) }
}
